classes/pdf/HTMLTemplateOrderSlip.php

```diff
@@ -67,11 +67,31 @@ public function __construct(OrderSlip $order_slip, Smarty $smarty)
         // header informations
         $this->date = Tools::displayDate($this->order_slip->date_add);
         $prefix = Configuration::get('PS_CREDIT_SLIP_PREFIX', Context::getContext()->language->id);
-        $this->title = sprintf(HTMLTemplateOrderSlip::l('%1$s%2$06d'), $prefix, (int) $this->order_slip->id);
+        $this->title = sprintf(HTMLTemplateOrderSlip::l('%1$s%2$06d'), $prefix, $this->getNumberOrderSlip((int) $order_slip->id));
 
         $this->shop = new Shop((int) $this->order->id_shop);
     }
 
+    /**
+     * Returns the numbering of the subscription invoice with respect to the year.
+     *
+     * @param int $id_order_slip order_slip
+     *
+     * @return int number for year
+     */
+    public function getNumberOrderSlip(int $id_order_slip): int
+    {
+        $temporary_order_slip = Db::getInstance()->executeS('SELECT date_add FROM `' . _DB_PREFIX_ . 'order_slip` WHERE id_order_slip=' . $id_order_slip);
+
+        $current_year = date('Y-01-01 00:00:00', strtotime($temporary_order_slip[0]['date_add']));
```

Comment:
You have to test if ` $temporary_order_slip[0]['date_add']` is available, same above

Please use camelCase instead of snake_case :)

Reply:
Hi, and updated the code to perform the check and used camelCase. a greeting